File: DarkOrbit 10.0/Net/netty/commands/ActivatePortalCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class ActivatePortalCommand
    {
        public const short ID = 2310;

        public static byte[] write(int mapId, int portalId)
        {
            var param1 = new ByteArray(ID);
            param1.writeInt(mapId >> 3 | mapId << 29);
            param1.writeInt(portalId << 15 | portalId >> 17);
            param1.writeShort(20220);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/ActivatePortalCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class ActivatePortalCommand
        implements ServerCommand {

    public static int ID       = 30733;
    public        int mapId    = 0;
    public        int var_3576 = 0;

    public ActivatePortalCommand(int param1, int param2) {
        this.mapId = param1;
        this.var_3576 = param2;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.var_3576 << 12 | this.var_3576 >>> 20);
            param1.writeInt(this.mapId << 13 | this.mapId >>> 19);
            param1.writeShort(22459);
        } catch (IOException e) {
        }
    }
}